### PR TITLE
docs: move the test readme to the official doc

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

The instructions to deploy a developer environment are useful and should
be part of the official documentation instead of being hidden in the
repo's README.md. Also, remove ancient documentation to deploy with
kubespray a multi-node env.
Also, add a warning to always run tests env in a virtual machine.

Closes: https://github.com/rook/rook/issues/9166
Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by subhamkrai — APPROVED** [review]
LGTM

File: Documentation/development-environment.md [hunk 1]
```diff
@@ -1,226 +1,58 @@
 ---
-title: Multi-Node Test Environment
+title: Developer Environment
 weight: 12100
```
Comment (travisn) [inline Documentation/development-environment.md:3]: How about moving this before the storage providers doc?
```suggestion
weight: 12030
```

File: Documentation/development-environment.md [hunk 2]
```diff
@@ -1,226 +1,58 @@
...
-* [Using CodeReady Containers for setting up single node openshift 4.x cluster](#using-codeready-containers-for-setting-up-single-node-openshift-4x-cluster)
+You can choose any Kubernetes install of your choice. The test framework only depends on `kubectl`
+being configured. To install `kubectl`, please see the [official guide](https://kubernetes.io/docs/tasks/tools/#kubectl).
 
-## Using KVM/QEMU and Kubespray
+# Minikube
 
-### Setup expectation
+The developers of Rook are working on Minikube and thus it is the recommended way to quickly get
+Rook up and running. The Rook authors do not recommend using Minikube for production but
```
Comment (travisn) [inline Documentation/development-environment.md:15]:
```suggestion
Rook up and running. Minikube should not be used for production but the Rook authors
```

**Review by travisn — CHANGES_REQUESTED** [review]
Finally we get rid of the old multi-node test topic and move the dev guide to the doc. Thanks!

**Review by travisn — APPROVED** [review]

**Pull request merged**